pkg/autodiscovery/listeners/kubelet.go

```diff
@@ -280,6 +292,30 @@ func podHasADTemplate(annotations map[string]string, containerName string) bool
 	return false
 }
 
+// getCheckNamesFromAnnotations unmarshals the json string of check names
+// defined in pod annotations and returns a slice of check names
+func getCheckNamesFromAnnotations(annotations map[string]string, containerName string) ([]string, error) {
+	if checkNamesJSON, found := annotations[fmt.Sprintf(newPodAnnotationCheckNamesFormat, containerName)]; found {
+		checkNames := []string{}
+		err := json.Unmarshal([]byte(checkNamesJSON), &checkNames)
+		if err != nil {
+			log.Errorf("Cannot parse check names: %v", err)
+			return nil, err
```

Comment:
I think it is better to log the error in the parent function, that ignore the error.

```suggestion
			return nil, fmt.Errorf("Cannot parse check names: %v", err)
```